services/distribution/src/main/java/app/coronawarn/server/services/distribution/dgc/client/CloudDccFeignHttpClientProvider.java

```diff
@@ -0,0 +1,95 @@
+package app.coronawarn.server.services.distribution.dgc.client;
+
+import app.coronawarn.server.common.federation.client.hostname.HostnameVerifierProvider;
+import app.coronawarn.server.services.distribution.config.DistributionServiceConfig;
+import feign.Client;
+import feign.httpclient.ApacheHttpClient;
+import java.io.File;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.springframework.cloud.commons.httpclient.ApacheHttpClientConnectionManagerFactory;
+import org.springframework.cloud.commons.httpclient.ApacheHttpClientFactory;
+import org.springframework.cloud.commons.httpclient.DefaultApacheHttpClientConnectionManagerFactory;
+import org.springframework.cloud.commons.httpclient.DefaultApacheHttpClientFactory;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Component;
+
+/**
+ * Creates a dedicated http client used by Feign when performing http calls to the Federation Gateway Service.
+ */
+@Component
+@Profile("!fake-dcc-client")
+public class CloudDccFeignHttpClientProvider implements DccFeignHttpClientProvider {
+
+  private final Integer connectionPoolSize;
+  //  private final File keyStore;
+  //  private final String keyStorePassword;
+  //  private final File trustStore;
+  //  private final String trustStorePassword;
+
+  private final HostnameVerifierProvider hostnameVerifierProvider;
+
+  /**
+   * Construct Provider.
+   *
+   * @param config .
+   */
+  public CloudDccFeignHttpClientProvider(DistributionServiceConfig config,
+      HostnameVerifierProvider hostnameVerifierProvider) {
+    //    var ssl = config.getDigitalGreenCertificate().getClient().getSsl();
+    this.connectionPoolSize = config.getConnectionPoolSize();
+    //    this.keyStore = ssl.getKeyStore();
+    //    this.keyStorePassword = ssl.getKeyStorePassword();
+    //    this.trustStore = ssl.getTrustStore();
+    //    this.trustStorePassword = ssl.getTrustStorePassword();
+    this.hostnameVerifierProvider = hostnameVerifierProvider;
+  }
+
+  /**
+   * Creates a FeignClient.
+   */
+  @Override
+  @Bean
+  public Client createFeignClient() {
+    return new ApacheHttpClient(
+        federationHttpClientFactory(connectionPoolSize).createBuilder().build());
```

Comment:
@AliceBilc - if you're reading into the feign client stuff... when are the opened sockets actually released?